### PR TITLE
sort violations to avoid unnecessary diffs in the baseline

### PR DESCRIPTION
**Pull request by dbu**
in our project, we noticed that the order of entries when generating the baseline varies randomly. 

this changes it to sort all entries in the baseline, making it possible to see what the actual changes in the baseline are.

### CODE REVIEW

**Review by AlessandroMinoccheri — APPROVED** [review]
LGTM 👍

**Comment by AlessandroMinoccheri** [comment]
@dbu can you please fix the CI?

**Comment by dbu** [comment]
there seems to be a problem with the php 7 builds (phpunit version). i will look at the test fixtures as per the php 8.1 build (i guess the ordering messes with the expected message).

File: tests/E2E/Cli/CheckCommandTest.php [hunk 1]
```diff
@@ -48,8 +48,8 @@ public function test_app_returns_error_with_multiple_violations(): void
   should implement ContainerAwareInterface because all controllers should be container aware
 
 App\Domain\Model has 2 violations
-  depends on App\Services\UserService, but should not depend on classes outside namespace App\Domain because we want protect our domain (on line 14)
-  depends on App\Services\CartService, but should not depend on classes outside namespace App\Domain because we want protect our domain (on line 15)';
+  depends on App\Services\CartService, but should not depend on classes outside namespace App\Domain because we want protect our domain (on line 15)
+  depends on App\Services\UserService, but should not depend on classes outside namespace App\Domain because we want protect our domain (on line 14)';
```
Comment (dbu) [inline tests/E2E/Cli/CheckCommandTest.php:52]: this shows something interesting. the `<=>` checks things in order as the properties appear in the class (afaik), so the error message is "more important" than the line number.

File: src/Rules/Violation.php [hunk 2]
```diff
@@ -9,12 +9,12 @@ class Violation implements \JsonSerializable
...
     private $fqcn;
 
-    /** @var string */
-    private $error;
-
     /** @var int|null */
     private $line;
 
+    /** @var string */
+    private $error;
```
Comment (dbu) [inline src/Rules/Violation.php:16]: this is the easiest way to get the sorting with `<=>` right. i think it makes more sense to have fqcn -> line number -> error.

File: tests/E2E/Cli/CheckCommandTest.php (hunk 1 above)
Comment (dbu) [reply tests/E2E/Cli/CheckCommandTest.php]: i reverted this change and changed the Violation class to have the error after the line number.

**Comment by dbu** [comment]
> there seems to be a problem with the php 7 builds (phpunit version)

i misread the output, that was warnings and skipped tests. the failures were always about the order afaict

File: src/Rules/Violations.php [hunk 3]
```diff
@@ -117,6 +117,13 @@ static function (Violation $a, Violation $b): int {
         ));
     }
 
+    public function sort(): void
```
Comment (AlessandroMinoccheri) [inline src/Rules/Violations.php:120]: could it be useful to add a simple test to document well this new behavior?
Do you think is useful @dbu ?

**Review by AlessandroMinoccheri — CHANGES_REQUESTED** [review]

**Review by fain182 — APPROVED** [review]

File: src/Rules/Violations.php (hunk 3 above)
Comment (dbu) [reply src/Rules/Violations.php:120]: yeah, agree that we should have a test for this. i added the test now.

**Review by AlessandroMinoccheri — APPROVED** [review]

**Comment by AlessandroMinoccheri** [comment]
Thanks a lot @dbu !

**Pull request merged**